s2s/time.py

```diff
@@ -418,9 +410,18 @@ def __str__(self):
         return f"{self._n_intervals} periods of {self.freq} leading up to {self.month}/{self.day}."
 
     def __repr__(self):
+        if self._intervals is not None:
+            return str(self._intervals)
```

Comment:
In notebook the calendar table looks a bit weird after mapping years to the calendar due to this brute-force conversion:

```py
i_interval                         0                         1   \
anchor_year                                                       
2021         (2021-09-15, 2021-10-15]  (2021-08-16, 2021-09-15]   
2020         (2020-09-15, 2020-10-15]  (2020-08-16, 2020-09-15]   
2019         (2019-09-15, 2019-10-15]  (2019-08-16, 2019-09-15]   

i_interval                         2                         3   \
anchor_year                                                       
2021         (2021-07-17, 2021-08-16]  (2021-06-17, 2021-07-17]   
2020         (2020-07-17, 2020-08-16]  (2020-06-17, 2020-07-17]   
2019         (2019-07-17, 2019-08-16]  (2019-06-17, 2019-07-17] 
```
Actually `__repr__` is supposed to return strings in practice, rather than a table. I think we'd better not use it this way. Since the 2D table view of the calendar is always a good feedback to the user, it is ok to make it visible `self.intervals`.

Reply:
Ah, my bad! We should return the __repr__ method of the pandas dataframe instead

Reply:
```suggestion
        if self._intervals is not None:
            return repr(self._intervals)
```

Doest that make it better?

Reply:
😓 Not really...It still looks the same. I think internally `repr(object)` tries to convert things to strings. The display of dataframe looks ok in jupyter notebook because it is treated as a html object (I learn this from google...). So the conversion to strings does not work.

Reply:
Fixed by adding a `def _repr_html_(self):` function specifically for jupyter notebooks.

The solution is found [here](https://stackoverflow.com/questions/32848845/what-pandas-dataframe-method-tells-ipython-notebook-to-display-as-html).